### ISSUE TITLE
Dividing by a single quoted numeric literal has different results than an unquoted numeric literal

### ISSUE THREAD

**Issue by elonazoulay** [apache/pinot#11676]
It appears that Pinot 1.0.0 treats quoted numeric literals differently than unquoted numeric literals:

The following query will return `Long.MIN_VALUE`:
```
SELECT <column>/'3600' FROM <table>
```

The following query will return results as expected:
```
SELECT <column>/3600 FROM <table>
```

**Comment by Jackie-Jiang** [comment]
Quoted numeric literal is parsed as string, and pinot might throw exception. Do you see pinot return `Long.MIN_VALUE` explicitly?

**Comment by elonazoulay** [comment]
Yes, during parsing. I don't see exceptions. This differs from previous versions where numeric literal was treated the same regardless of single quoting. Lots of tools also single quote all literals as well, regardless of type, this appears to be new.

**Comment by Jackie-Jiang** [comment]
The behavior is indeed changed with #10380 and we marked it backward incompatible. Currently string literal will be treated as 0 when used as number. We need to revisit the literal type handling and make a fix.

**Comment by Jackie-Jiang** [comment]
This can be fixed with #11749

**Comment by Jackie-Jiang** [comment]
Fixed with #11763
